jssource/src_files/include/javascript/ajaxUI.js

```diff
@@ -168,14 +168,26 @@ SUGAR.ajaxUI = {
     {
         if(YAHOO.lang.trim(url) != "")
         {
+
+          // ajaxUILoc XSS protection:
+          // window.location = url; is vulnerable to XSS attack
+          // Check for valid url
+          // Expects url encoded versions of index.php?module=Home&action=index&parentTab=All
+          var testUrl = decodeURIComponent(url);
+          if (
+            /^index.php([A-Z]|[%=+#&\?\[\]\.]|[a-z]|\d|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])+/i.test(testUrl) === false
```

Comment:
A little bit more specific and strict regex:
`/^index.php?(([A-Z]|[a-z]|\d|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])+[%=+#&\[\]\.]*)+/i.test(testUrl)`